### PR TITLE
Create a Quickstart to run a Tekton task with Minikube

### PR DESCRIPTION
**Pull request by geriom**
# Changes

Introductory guide friendly for newcomers. The approach in this doc is to be
easier to follow by providing prescriptive instructions about the tools
to use. After installing the prerequisites, doing copy/paste of the code
samples should work.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/website/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by jjasghar** [comment]
Taking a look today, give me a bit of time to read over this.

File: content/en/docs/Getting Started/quickstart-kind.md
Comment (bobcatfish) [inline content/en/docs/Getting Started/quickstart-kind.md:82]: typo taks -> task

File: content/en/docs/Getting Started/quickstart-kind.md
Comment (bobcatfish) [inline content/en/docs/Getting Started/quickstart-kind.md:101]: there's something odd about the indentation here - i think it's way to deep (2 spaces would probably be good, or 4 at most?) and it shows up as red in the preview 🤔 

image

File: content/en/docs/Getting Started/quickstart-kind.md
Comment (bobcatfish) [inline content/en/docs/Getting Started/quickstart-kind.md:121]: probably makes sense to use TaskRuns explicitly but just so you know there's one other option that might be a nice user experience - but the user would have to also install `tkn` (https://github.com/tektoncd/cli) so maybe it's not worth the extra setup - but if it wasn't too much trouble, instead of making a TaskRun here the user could do [`tkn task start echo-hello-world`](https://github.com/tektoncd/cli/blob/main/docs/cmd/tkn_task_start.md)

File: content/en/docs/Getting Started/quickstart-kind.md
Comment (bobcatfish) [inline content/en/docs/Getting Started/quickstart-kind.md:156]: instead of showing them the pod, it might be a bit nicer to show them the taskrun, e.g. `kubectl --context kind-tektoncd get taskrun echo-hello-world-task-run` would show something simple like:

```
NAME                               SUCCEEDED    REASON       STARTTIME   COMPLETIONTIME
echo-hello-world-task-run          True         Succeeded    22h         22h
```

plus the pods are (kind of) an implementation detail of tekton anyway 🤞  (tho in reality is not like ppl can really ignore them...)

**Comment by jjasghar** [comment]
My only question, is this to show off Tekton _inside_ of KiND? Otherwise I'm not seeing the advantage of this over https://tekton.dev/docs/getting-started/pipelines/

**Review by bobcatfish — COMMENTED** [review]
Very cool!! I was able to run through it myself so easily (i might be not quite the target audience tho haha - BUT ive never really used KIND before so that was pretty cool for me!)

Couple of thoughts:
* This is kind of a nit but I'm wondering if we want to have this repo as the source of truth for this quickstart - it seems like it's only using Tekton Pipelines, would it make sense to store it (and potentially version and update it) alongside the Tekton Pipelines docs? (maybe it could even replace [the tutorial over there?)](https://github.com/tektoncd/pipeline/blob/main/docs/tutorial.md)
* What do you think about updating or maybe replacing the existing getting started landing page? (https://tekton.dev/docs/getting-started/) (seems like this kind tutorial would be a good replacement?)

**Comment by bobcatfish** [comment]
@jjasghar i think what's really nice about using kind is that it's very easy to setup and lets us have getting started instructions that are pretty cloud agnostic (but still tell you how to setup the cluster in the first place!)

my understanding is that this tutorial _would_ replace at least 1 (if not more - we have some duplication going on) of our existing tutorials. the one you linked to (https://tekton.dev/docs/getting-started/pipelines/) i think is meant to build on a previous task based tutorial and add pipelines? (presumably it would work on the kind cluster as well? maybe needs a few updates to flow nicely from the kind tutorial).  i think this tutorial would probably replace https://tekton.dev/docs/getting-started/ (maybe we go back to our old approach and have a tab or something for the openshift specific instructions?)

**Comment by geriom** [comment]
@jjasghar This is intended to be the very first Tekton doc for someone with no prior experience. A 5-minute task that is easy to complete so you get that "feels good" moment that you need to explore further. That's why it's prescriptive about Kind, it's an effort to guarantee the task can be completed successfully with minimal effort. The existing getting started requires some cognitive work to follow.

@bobcatfish You are correct, my intention is to consolidate the doc in the pipelines repo (https://github.com/tektoncd/pipeline/blob/main/docs/tutorial.md), the 2 getting started docs from tekton.dev (getting-started and getting-started/pipelines), and the Katacoda tuto (https://github.com/tektoncd/website/tree/main/tutorials/katacoda/getting-started) in to a single doc that we can link from everywhere. The quickstart in his PR being the entry point and a further more in-depth doc as the natural follow-up.


I'm going to take a look at the other comments and apply the fixes later this week. Thanks for the feedback.

**Comment by afrittoli** [comment]
Thanks @geriom - I haven't reviewed the instructions in details yet - but I definitely support using `kind` - that's what I use for my development work and POCs anyways. We have a script for that so perhaps we could include a link to it? https://github.com/tektoncd/plumbing/blob/main/hack/tekton_in_kind.sh

File: content/en/docs/Getting Started/quickstart-kind.md
Comment (geriom) [reply content/en/docs/Getting Started/quickstart-kind.md]: Fixed

File: content/en/docs/Getting Started/quickstart-kind.md
Comment (geriom) [reply content/en/docs/Getting Started/quickstart-kind.md]: I think these were tabs instead of spaces (?). This looked fine on my computer :). I made sure they were converted to spaces, I think it's fixed now.

File: content/en/docs/Getting Started/quickstart-kind.md
Comment (geriom) [reply content/en/docs/Getting Started/quickstart-kind.md:121]: I thought about it, same to check the task status. The trade-off of using a simpler command but having to install something extra, if this is your first ever experience with tekton, may not be worth it in my opinion. For the longer tutorial covering pipelines in more detail it's definitely a good idea to ask for the `tkn` client prerequisite.

File: content/en/docs/Getting Started/quickstart-kind.md
Comment (geriom) [reply content/en/docs/Getting Started/quickstart-kind.md]: Updated

**Comment by geriom** [comment]
> * This is kind of a nit but I'm wondering if we want to have this repo as the source of truth for this quickstart - it seems like it's only using Tekton Pipelines, would it make sense to store it (and potentially version and update it) alongside the Tekton Pipelines docs? (maybe it could even replace [the tutorial over there?)](https://github.com/tektoncd/pipeline/blob/main/docs/tutorial.md)

If we move it to that repo it would be harder to make it a landing page because of how the sync script works.

> * What do you think about updating or maybe replacing the existing getting started landing page? (https://tekton.dev/docs/getting-started/) (seems like this kind tutorial would be a good replacement?)

I like the idea. Should I proceed and move the current content to a different file?

**Comment by bobcatfish** [comment]
> If we move it to that repo it would be harder to make it a landing page because of how the sync script works.

😨  well maybe in the meantime we could link directly to the page on tekton.dev from the pipelines repo :S

> Should I proceed and move the current content to a different file?

I'd remove the existing file entirely - in the end I think our docs at tekton.dev should be pretty authorative vs. kind of a grab bag of different tutorials. The tutorial will also still be in version control if someone wants to resurrect it later.

**Comment by bobcatfish** [comment]
... did not mean to hit that button

**Comment by bobcatfish** [comment]
TOO MUCH POWER

**Comment by afrittoli** [comment]
Waiting for the doc plan to be agreed on, then we can resume this one and get it merged

**Comment by geriom** [comment]
@afrittoli This PR is ready to be merged after a a second round of reviews.

Updates:

I switched from Kind to Minikube because of the recent changes in the Docker licence. With Minikube the users can pick a different engine and are not forced to use docker to follow the instructions.

I also added a second page, this is now a two-part tutorial, somewhat mirroring the content we had before:

- Create a task
- Create a second task and a pipeline.

I also deleted the original tutorial that we had in `_index` and made it a simple landing page.

**Comment by AlanGreene** [comment]
Looks like this removes reference to the nightly releases. It would be good to restore that as we do want to make them more visible. Ref: https://github.com/tektoncd/website/pull/344#issuecomment-1038992697

**Comment by geriom** [comment]
In the spirit of the [new guidelines for tutorials](https://tekton.dev/docs/contribute/doc-con-content/), I think we should pick just one path and go with it throughout the entire doc, don't cover alternatives in the getting started. Instead, I would prefer to create an installation doc, that covers all the available installation options.

File: content/en/docs/Getting Started/tasks.md
Comment (bobcatfish) [inline content/en/docs/Getting Started/tasks.md:90]: nit: maybe instead of "build work" something more generic (but CI/CD specific), something like "perform logic the Task requires" or drop that part of the sentence entirely? (cuz not every Task is doing something build related)

File: content/en/docs/Getting Started/tasks.md
Comment (bobcatfish) [inline content/en/docs/Getting Started/tasks.md:156]: nit: might be nice to also look at the logs at this point, something like `k logs --selector=tekton.dev/taskRun=hello-task-run` but np if that's just complicating things!

File: content/en/docs/Getting Started/pipelines.md
Comment (bobcatfish) [inline content/en/docs/Getting Started/pipelines.md:45]: there's some extra whitespace at the beginning of the yaml block that shows up when i try to copy paste from this block:

image

File: content/en/docs/Getting Started/pipelines.md
Comment (bobcatfish) [inline content/en/docs/Getting Started/pipelines.md:81]: should be `taskRef`

(i think this will work initially but it leads to really weird bugs X'D https://github.com/tektoncd/triggers/issues/526)

**Review by bobcatfish — COMMENTED** [review]
very nice!!! just some minor comments from me, that was very smooth for me (and the first time ive used minikube in... years XD might do it more often now!)

(i'm going to hold off on doing an approve or lgtm and defer to folks that have been attending the docs WG but lemme know if I need to)

File: content/en/docs/Getting Started/tasks.md
Comment (geriom) [reply content/en/docs/Getting Started/tasks.md:156]: Updated

File: content/en/docs/Getting Started/pipelines.md
Comment (geriom) [reply content/en/docs/Getting Started/pipelines.md:81]: Fixed!

**Review by afrittoli — COMMENTED** [review]
Thanks for this - I added some cosmetic comments that needs to be fixed before this is merged, but the content looks great.

Historically with always been starting our Tekton guides with writing your own `Task` and `Pipeline`... I wonder however if we could rethink that as an "advanced" step - and start with using `Tasks` from the catalog instead? Not something for this PR, but I wanted to bring up the idea. @tektoncd/core-maintainers thoughts?

**Comment by bobcatfish** [comment]
> Historically with always been starting our Tekton guides with writing your own Task and Pipeline... I wonder however if we could rethink that as an "advanced" step - and start with using Tasks from the catalog instead? Not something for this PR, but I wanted to bring up the idea. https://github.com/orgs/tektoncd/teams/core-maintainers thoughts?

that could make a lot of sense @afrittoli  - could get users much more quickly to the satisfaction of running something and having it work :D

File: content/en/docs/Getting Started/tasks.md
Comment (geriom) [reply content/en/docs/Getting Started/tasks.md]: I updated the sample to use `script` instead.,

File: content/en/docs/Getting Started/pipelines.md
Comment (geriom) [reply content/en/docs/Getting Started/pipelines.md]: Somehow I ended up with tabs instead of spaced. It should be fixed now.

File: content/en/docs/Getting Started/tasks.md
Comment (geriom) [reply content/en/docs/Getting Started/tasks.md:120]: All the rendering glitches can be fixed using a newer version of hugo. #361 Fixes this.

File: content/en/docs/Getting Started/tasks.md
Comment (geriom) [reply content/en/docs/Getting Started/tasks.md:76]: Fixed by #361

**Comment by geriom** [comment]
Completely agree with both of you @bobcatfish and @afrittoli . I'm going to give it some thought and share my ideas in either the WG or the mailing list.

**Comment by afrittoli** [comment]
@geriom @AlanGreene I kicked off a re-deploy now, but it did not help - I still see the same cosmetic glitches in the rendering. I think this needs to be rebased so we can preview the actual result.

Apart from that the content looks good to me, so feel free to merge once the cosmetic issues are fixed

**Comment by geriom** [comment]
@afrittoli I rebased the branch, the preview looks fine now.

@AlanGreene I think we can merge it.

File: content/en/docs/Getting Started/tasks.md
Comment (AlanGreene) [inline content/en/docs/Getting Started/tasks.md:125]: similar indentation issue with this block when rendered

File: content/en/docs/Getting Started/tasks.md
Comment (geriom) [reply content/en/docs/Getting Started/tasks.md:20]: Good catch! That's is the right link, thank you.

File: content/en/docs/Getting Started/tasks.md
Comment (geriom) [reply content/en/docs/Getting Started/tasks.md:125]: Fixed

File: content/en/docs/Getting Started/tasks.md
Comment (geriom) [reply content/en/docs/Getting Started/tasks.md:176]: Fixed

File: content/en/docs/Getting Started/tasks.md
Comment (geriom) [reply content/en/docs/Getting Started/tasks.md]: Good catch! That is indeed the correct link

**Review by AlanGreene — APPROVED** [review]
Looks good, thanks @geriom 

/lgtm
/hold cancel
/meow

**Pull request merged**
